### PR TITLE
Add should-fail test for merge with relationship

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
@ejegg  - that getSingle should fail if 2 exist (currently not for me)

Before
----------------------------------------
_What is the old user-interface or technical-contract (as appropriate)?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

After
----------------------------------------
_What changed? What is new old user-interface or technical-contract?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

Technical Details
----------------------------------------
_If the PR involves technical details/changes/considerations which would not be manifest to a casual developer skimming the above sections, please describe the details here._

Comments
----------------------------------------
_Anything else you would like the reviewer to note_

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@ejegg this is now happily failing :-)

File: tests/phpunit/CRM/Dedupe/MergerTest.php [hunk 1]
```diff
@@ -1472,6 +1474,26 @@ public function testMergeWithDeclaredSearchJoin(): void {
...
+  /**
+   * Test that the presence of a custom field attached to the relationship does not block merge.
+   */
+  public function testMergeWithRelationshipWithCustomFields(): void {
+    $contact1 = $this->individualCreate();
+    $contact2 = $this->individualCreate();
+    $this->createCustomGroupWithFieldOfType(['extends' => 'Relationship']);
+    $this->callAPISuccess('Relationship', 'create', [
+      'contact_id_a' => $contact2,
+      'contact_id_b' => CRM_Core_Config::domainID(),
```
Comment (seamuslee001) [inline tests/phpunit/CRM/Dedupe/MergerTest.php:1486]: @eileenmcnaughton is this a proper contact id value?

Comment (eileenmcnaughton) [reply tests/phpunit/CRM/Dedupe/MergerTest.php:1486]: hmm accidentally yes.....

**Comment by eileenmcnaughton** [comment]
Correct fail

```
CRM_Dedupe_MergerTest::testMergeWithRelationshipWithCustomFields
Invalid getsingle resultArray
(
    [count] => 2
    [is_error] => 1
    [error_message] => Expected one Relationship but found 2
)

 entity: Relationship . 
 params 
 Array
(
    [contact_id_a] => 3
    [version] => 3

```

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
I added the fix to this so it should be passing now - but have lost Jenkins record of the last run

**Comment by eileenmcnaughton** [comment]
Additional relevant tests are:
```
api_v3_RelationshipTest::testRelationshipCreateDuplicateWithCustomFields2
  
api_v3_RelationshipTest::testRelationshipCreateDuplicateWithCustomFields3
```
  these go through the main `Relationship:create` function which seems to have different expectations to the `Relationship:legacyMultipleCreate` called by the employer update code (called from merge code). I think something weird is going on in `legacyMultipleCreate` so am attempting to divide & conquery via #22385

**Pull request closed without merging**